sdk/openai/Azure.AI.OpenAI/src/Custom/OpenAIClient.cs

```diff
@@ -42,5 +45,73 @@ public virtual Response<Completions> GetCompletions(string deploymentId, string
             return GetCompletions(deploymentId, completionsOptions, cancellationToken);
         }
+
+        public virtual Response<StreamingCompletions> GetStreamingCompletions(
+            string deploymentId,
+            CompletionsOptions completionsOptions,
+            CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNullOrEmpty(deploymentId, nameof(deploymentId));
+            Argument.AssertNotNull(completionsOptions, nameof(completionsOptions));
+
+            using DiagnosticScope scope = ClientDiagnostics.CreateScope(
+                "OpenAIClient.GetStreamingCompletions");
+            scope.Start();
+
+            RequestContext context = FromCancellationToken(cancellationToken);
+
+            RequestContent nonStreamingContent = completionsOptions.ToRequestContent();
+            RequestContent streamingContent = GetStreamingEnabledRequestContent(nonStreamingContent);
+
+            try
+            {
+                HttpMessage message = CreateGetCompletionsRequest(deploymentId, streamingContent, context);
+                message.BufferResponse = false;
+                Response baseResponse = _pipeline.ProcessMessage(message, context, cancellationToken);
+                return Response.FromValue(new StreamingCompletions(baseResponse), baseResponse);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        public virtual Task<Response<StreamingCompletions>> GetStreamingCompletionsAsync(
+            string deploymentId,
+            CompletionsOptions completionsOptions,
+            CancellationToken cancellationToken = default)
+        {
+            return Task.FromResult<Response<StreamingCompletions>>(null);
+        }
+
+        private static RequestContent GetStreamingEnabledRequestContent(RequestContent originalRequestContent)
+        {
+            // Dump the original request content to a temporary stream and seek to start
+            using Stream originalRequestContentStream = new MemoryStream();
+            originalRequestContent.WriteTo(originalRequestContentStream, new CancellationToken());
+            originalRequestContentStream.Position = 0;
+
+            JsonDocument originalJson = JsonDocument.Parse(originalRequestContentStream);
+            JsonElement originalJsonRoot = originalJson.RootElement;
+
+            Utf8JsonRequestContent augmentedContent = new Utf8JsonRequestContent();
+            augmentedContent.JsonWriter.WriteStartObject();
+
+            // Copy the original JSON content back into the new copy
+            foreach (JsonProperty jsonThing in originalJsonRoot.EnumerateObject())
+            {
+                augmentedContent.JsonWriter.WritePropertyName(jsonThing.Name);
+                jsonThing.Value.WriteTo(augmentedContent.JsonWriter);
+            }
+
+            // ...Add the *one thing* we wanted to add
```

Comment:
@annelo-msft, here is an interesting use case for your mutable JSON APIs. For context, this routine is rewriting request content just to add one property to the JSON payload.

Reply:
Love it!  Thank you, @KrzysztofCwalina!

Reply:
Agreed, that would be super duper nice. We'll hopefully be able to address our instance issue with an internal generated field or the like, but having a generalized capability to mutate payload in this partial class setup with custom/generated interrelationship would be fantastic.